### ISSUE TITLE
File-scoped `flake8: noqa` directive with comment is not recognized

### ISSUE THREAD

**Issue by tylerlaprade** [astral-sh/ruff#10239]
```
# flake8: noqa -- comment explaining why we're disabling flake8 for this file
```
As per [Ruff's documentation](https://docs.astral.sh/ruff/linter/#error-suppression), this is treated the same as `# ruff: noqa`, which doesn't allow comments. However, `flake8` _does_ allow comments - and I happen to believe comments should be required any time some/all rules are disabled to explain why. This particular comment is in fact being used to disable `flake8` for this file. The author would have liked to also disable Ruff for the whole file, but didn't realize that the comment could do both due to the comment, and ended up disabling the particular line with a Ruff violation.

This came to my attention when I upgraded to Ruff v0.3.0 (I'm very excited to try the new formatting, by the way!). I ran `ruff check` and got a warning every time:
```
warning: Invalid `# ruff: noqa` directive at abacus/migrations/0324_manual_recompute_all_patient_journeys.py:2: expected `:` followed by a comma-separated list of codes (e.g., `# noqa: F401, F841`).
```

Ideally, comments would be allowed on `ruff: noqa`. But even if not, Ruff should not be warning about a valid `flake8` directive.

**Comment by charliermarsh** [comment]
I think a format we could allow (which would be consistent with `# noqa`) would be like: `# flake8: noqa # some comment`.

**Comment by tylerlaprade** [comment]
I got in the habit of using `--` from ESLint, but any consistent format would be good, and I think it'd be good if Ruff chose a single format, whatever that is, to help the community standardize (and to encourage commenting on why rules are suppressed).

That being said, anything that Flake8 currently allows should at the very least not trigger a Ruff warning, since a large legacy codebase might be discouraged from migrating if they see hundreds of warnings on previously valid code.

**Comment by zanieb** [comment]
Related discussion at https://github.com/astral-sh/ruff/pull/8876#issuecomment-1831171627

**Comment by zanieb** [comment]
Going to track this in https://github.com/astral-sh/ruff/issues/10343 instead since this is not a flake8-specific problem.

Thank you for raising! If you have thoughts on the design please respond over there.
